### PR TITLE
Pract.01: Add 07-08 | Nikita Beliy

### PR DESCRIPTION
**Pull request by N1k1ta-white**
Nikita Beliy
SI 1 9MI800022

### CODE REVIEW

File: Task07.cpp [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+using std::endl;
+
+enum Genre {
+    Comedy,
+    Action,
+    Horror
+};
+
+struct Movie {
+    char name[100];
```
Comment (khatoni) [inline Task07.cpp]: It would be better to not use numbers. Instead use global constant like const int MAX_NAME_SIZE=100

File: Task07.cpp [hunk 2]
```diff
@@ -0,0 +1,60 @@
+#include <iostream>
+
+using std::cout;
+using std::cin;
+using std::endl;
+
+enum Genre {
```
Comment (khatoni) [inline Task07.cpp]: It would be better to use enum class

File: Task07.cpp [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    double rate;
+};
+
+void str_trans(char str1[], const char str2[]) {
+    int rotator = -1;
+    while (str2[++rotator] != '\0')
+        str1[rotator] = str2[rotator];
+}
+
+Movie add_movie(const char name[], int time, Genre gen, double rate) {
```
Comment (khatoni) [inline Task07.cpp]: const int , const double , const Genre

File: Task07.cpp [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+}
+
+void swap_movies(Movie &mov1, Movie &mov2) {
+    Movie agent;
+    agent = mov1;
+    mov1 = mov2;
+    mov2 = agent;
+}
+
+void sort_movies(Movie *movies, int num) {
```
Comment (khatoni) [inline Task07.cpp]: const int num

File: Task08.cpp [hunk 5]
```diff
@@ -0,0 +1,69 @@
+#include <iostream>
+
+using std::cout;
+using std::cin;
+using std::endl;
+
+struct Student {
+    char name[30];
```
Comment (khatoni) [inline Task08.cpp]: const int MAX_NAME_SIZE=30;
char name[MAX_NAME_SIZE+1];

File: Task08.cpp [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+
+Group create_group() {
+    int num;
+    double avarage_grade = 0;
+    cout << "Number of students: ";
+    cin >> num;
+    Group res{num};
+    for (int i = 0; i < num; ++i) {
+        cout << "Name of student " << i + 1 << ": ";
+        cin >> res.students[i].name;
```
Comment (khatoni) [inline Task08.cpp]: This way you cannot enter first name and last name;
Use getline
When you enter from the console char* and then you want to enter number you need to use cin.ignore();

File: Task08.cpp [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+        cin >> res.students[i].fac_num;
+        cout << "Grade of student " << i + 1 << ": ";
+        cin >> res.students[i].grade;
+        avarage_grade += res.students[i].grade;
+    }
+    res.avarage_grade = avarage_grade / num;
+    return res;
+}
+
+int how_many_students_get_money(const Group& group, double min_grade) {
```
Comment (khatoni) [inline Task08.cpp]: const double min_grade

File: Task08.cpp [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+    int res = 0;
+    for (int i = 0; i < group.number_of_students; ++i) {
+        if (group.students[i].grade < min_grade)
+            continue;
+        res++;
+    }
+    return res;
+}
+
+Group students_with_money(const Group& group, double min_grade) {
```
Comment (khatoni) [inline Task08.cpp]: const double min_grade

File: Task08.cpp [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+    int pos = 0;
+    double avarage_grade = 0;
+    Group res{how_many_students_get_money(group, min_grade)};
+    for (int i = 0; i < group.number_of_students; ++i) {
+        if (group.students[i].grade < min_grade)
+            continue;
+        res.students[pos++] = group.students[i];
+        avarage_grade += group.students[i].grade;
+    }
```
Comment (khatoni) [inline Task08.cpp]: The task is to sort the students who will get scholarship

**Review by khatoni — CHANGES_REQUESTED** [review]

**Comment by Justsvetoslavov** [comment]
Good solutions. Good job 👍

**Pull request closed without merging**